### PR TITLE
1085 add new connection setting

### LINKED ISSUE #1085: Add new connection setting
https://github.com/flanksource/duty/blob/main/schema/connections.hcl

### PR DESCRIPTION
**Pull request by Prashanth-Nelli**
Closes #1085

### CODE REVIEW

File: src/components/Connections/connectionTypes.tsx [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+        label: "URL",
+        key: "url",
+        type: fieldTypes.input,
+        required: true
+      },
+      {
+        label: "Username",
+        key: "username",
+        type: fieldTypes.EnvVarSource,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:184]:
```suggestion
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+      },
+      {
+        label: "Password",
+        key: "password",
+        type: fieldTypes.EnvVarSource,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:190]:
```suggestion
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+        label: "URL",
+        key: "url",
+        type: fieldTypes.input,
+        required: true
+      },
+      {
+        label: "Username",
+        key: "username",
+        type: fieldTypes.EnvVarSource,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:324]:
```suggestion
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+      },
+      {
+        label: "Password",
+        key: "password",
+        type: fieldTypes.EnvVarSource,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:330]:
```suggestion
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+      },
+      {
+        label: "Domain",
+        key: "domain",
+        type: fieldTypes.input,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:365]:
```suggestion
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 6]
```diff
@@ -0,0 +1,561 @@
...
+        label: "Password",
+        key: "password",
+        type: fieldTypes.input,
+        required: true
+      },
+      {
+        label: "Port",
+        key: "port",
+        type: fieldTypes.input,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:436]:
```suggestion
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 7]
```diff
@@ -0,0 +1,561 @@
...
+        label: "Name",
+        key: "name",
+        type: fieldTypes.input,
+        required: true
+      },
+      {
+        label: "Region",
+        key: "region",
+        type: fieldTypes.input,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:470]:
```suggestion
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 8]
```diff
@@ -0,0 +1,561 @@
...
+        label: "Access Key",
+        key: "username",
+        type: fieldTypes.EnvVarSource,
+        required: true
+      },
+      {
+        label: "Secret Key",
+        key: "password",
+        type: fieldTypes.EnvVarSource,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:488]:
```suggestion
        label: "Profile",
        key: "profile",
        type: fieldTypes.input,
        required: false
      },
      {
        label: "Access Key",
        key: "username",
        type: fieldTypes.EnvVarSource,
        required: false
      },
      {
        label: "Secret Key",
        key: "password",
        type: fieldTypes.EnvVarSource,
        required: false
```

File: src/components/Connections/connectionTypes.tsx [hunk 9]
```diff
@@ -0,0 +1,561 @@
...
+        key: "name",
+        type: fieldTypes.input,
+        required: true
+      },
+      {
+        label: "Certificate",
+        key: "certificate",
+        type: fieldTypes.EnvVarSource,
+        variant: variants.large,
+        required: true
```
Comment (moshloop) [inline src/components/Connections/connectionTypes.tsx:531]:
```suggestion
        required: false
```

**Comment by moshloop** [comment]
@Prashanth-Nelli The AWS and Windows connections cannot be added

**Comment by Prashanth-Nelli** [comment]
> @Prashanth-Nelli The AWS and Windows connections cannot be added

issue is resolved

**Pull request merged**